aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,226 @@
+            Error message if validation fails, None otherwise.
+        """
+
+        # Check if  structure is given as it is required
+        for key in "structure":
+            if key not in value:
+                return f"required value was not provided for the `{key}` namespace."
+
+        return None
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files from the input nodes passed to this instance of the `CalcJob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        # Create needed inputs
+        # Define architecture from model if model is given, otherwise get architecture from inputs and download default model
+        architecture = (
+            str((self.inputs.model).architecture)
+            if self.inputs.model
+            else str(self.inputs.architecture.value)
+        )
+        model_path = (
+            str((self.inputs.model).filepath)
+            if self.inputs.model
+            else ModelData.download(
+                "http://tinyurl.com/46jrkm3v", architecture
+            ).filepath
+        )
+
+        # The inputs are saved in the node, but we want their value as a string
+        calctype = str((self.inputs.calctype).value)
+        precision = str((self.inputs.precision).value)
+        device = str((self.inputs.device).value)
+        xyz_filename = str((self.inputs.xyzoutput).value)
+        input_filename = self.inputs.metadata.options.input_filename
+
+        # Transform the structure data in cif file called input_filename
+        structure = self.inputs.structure
+        cif_structure = structure.get_cif()
+        with folder.open(self._DEFAULT_INPUT_FILE, "w") as inputfile:
+            inputfile.write(cif_structure.get_content())
+
+        cmd_line = {
+            "arch": architecture,
+            "structure": str(input_filename),
+            "device": device,
+            "calc-kwargs": {"model": model_path, "default_dtype": precision},
+            "write-kwargs": {"filename": xyz_filename},
+        }
+
+        codeinfo = datastructures.CodeInfo()
+
+        # Initialize cmdline_params as an empty list
+        codeinfo.cmdline_params = []
+        # Adding command line params for when we run janus
+        codeinfo.cmdline_params.append(calctype)
+        for flag, value in cmd_line.items():
+            if isinstance(value, dict):
+                # If the value is a dictionary, format it properly
+                formatted_value = (
+                    "{" + ", ".join([f"'{k}': '{v}'" for k, v in value.items()]) + "}"
+                )
```

Comment:
This is just a `repr` of the `dict`?

Reply:
`join` works with generators
```suggestion
                formatted_value = (
                    "{" + ", ".join(f"'{k}': '{v}'" for k, v in value.items()) + "}"
                )
```